spec/lib/roo/base_spec.rb

```diff
@@ -154,9 +154,14 @@ def sheets
     end
 
     context 'without a matching header row' do
-      it 'raises an error' do
+      it 'raises an error ' do
         expect { spreadsheet.row_with([/Missing Header/]) }.to \
-          raise_error(Roo::HeaderRowNotFoundError)
+          raise_error(Roo::HeaderRowNotFoundError, /Missing Header/)
+      end
+
+      it 'returns missing headers' do
+        expect { spreadsheet.row_with([/Header/, /Missing Header 1/, /Missing Header 2/]) }.to \
+          raise_error(Roo::HeaderRowNotFoundError, /Missing Header 1|Missing Header 2/)
```

Comment:
`/Missing Header 1|Missing Header 2/` only ensures that one of missing header is part of exception. 

If I change the code to raise just `missing_headers.first` this test case will still pass. I think this regex should ensure that all missing header/query is part of exception.

Reply:
agree, changed that!